File: LegendEditor/Assets/Scripts/Game/SceneMap/SceneMap.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using UnityEngine;

public class SceneMap : GameBase
{
	public MapHeader mHeader;
	public MapTile[] mTileList;
	public SceneMap()
	{
		mHeader = new MapHeader();
	}
	public void readFile(string fileName)
	{
		byte[] fileBuffer = null;
		FileUtility.openFile(fileName, ref fileBuffer);
		if(fileBuffer == null)
		{
			return;
		}
		int offset = 0;
		mHeader.parseHeader(fileBuffer, ref offset);
		int tileCount = mHeader.mWidth * mHeader.mHeight;
		mTileList = new MapTile[tileCount];
		for(int i = 0; i < tileCount; ++i)
		{
			mTileList[i] = new MapTile(i);
			mTileList[i].parseTile(fileBuffer, ref offset);
		}
	}
}

File: LegendDataEditor/LegendDataEditor/SceneMap/SceneMap.cpp
#include "SceneMap.h"
#include "MapTile.h"
#include "txSerializer.h"
#include "MapHeader.h"

SceneMap::SceneMap()
{
	TRACE_NEW(MapHeader, mHeader);
	mTileList = NULL;
}

void SceneMap::destroy()
{
	TRACE_DELETE(mTileList);
	TRACE_DELETE(mHeader);
}

void SceneMap::readFile(const string& fileName)
{
	int fileSize = 0;
	char* fileBuffer = FileUtility::openBinaryFile(fileName, &fileSize);
	int offset = 0;
	mHeader->parseHeader(fileBuffer, fileSize, offset);
	int tileCount = mHeader->mWidth * mHeader->mHeight;
	TRACE_NEW_ARRAY(MapTile, tileCount, mTileList);
	for (int i = 0; i < tileCount; ++i)
	{
		mTileList[i].parseTile(fileBuffer, fileSize, offset);
	}
	TRACE_DELETE_ARRAY(fileBuffer);
}